### ISSUE TITLE
Accessibility Violations found in carbon components

### ISSUE THREAD

**Issue by Pasoon** [carbon-design-system/ibm-security#772]
## Description 
Resolve accessibility violations in carbon components that were found in using the Accessibility Checker for chrome (https://www.ibm.com/able/toolkit/tools/)

**Report Screenshot**
![image](https://user-images.githubusercontent.com/17908476/95102900-f31c1e00-0701-11eb-8a5d-2cc4ef6cad9b.png)

The following violations were all found in one scan done on the Rules Explorer Page alone. There are several of the same violations in different pages across UCM. 

### Details

**Rule Violated:** The WAI-ARIA property must reference a non-empty unique id of an existing element that is visible
![image](https://user-images.githubusercontent.com/17908476/95101922-bb60a680-0700-11eb-8e08-dec161f7d013.png)

**Elements that Violated it:**

`<button aria-describedby="menu" aria-label="Main menu" class="security--button--icon security--toolbar__button security--button--icon--tooltip security--button--icon--tooltip--right">`

`<div tabindex="-1" aria-labelledby="downshift-0-label" role="listbox" class="bx--list-box__menu" id="downshift-0-menu">`

`<div tabindex="-1" aria-labelledby="downshift-1-label" role="listbox" class="bx--list-box__menu" id="downshift-1-menu">
`

**Rule Violated:** Each element with a "region" role must have a unique label
![image](https://user-images.githubusercontent.com/17908476/95102199-198d8980-0701-11eb-8db1-db1292bbe3c5.png)

**Elements that Violated it:**
`<section aria-label="" class="security--tag-wall">`

**Rule Violated:** Interactive component must have a programmatically associated name
![image](https://user-images.githubusercontent.com/17908476/95103473-a258f500-0702-11eb-8141-b18bb426b2e9.png)

**Elements that Violated it:**

`<div aria-expanded="false" tabindex="0" role="button" class="bx--tooltip__trigger">`

**Rule Violated:** Accessible name must match or contain the visible label text
![image](https://user-images.githubusercontent.com/17908476/95103770-0a0f4000-0703-11eb-82fd-041f8ac82556.png)

**Elements that Violated it:**

`<button tabindex="0" aria-label="Menu" class="bx--overflow-menu" aria-expanded="false" aria-haspopup="true">`

**Rule Violated:** Each element with "search" role must have a unique label that describes its purpose
![image](https://user-images.githubusercontent.com/17908476/95104057-62ded880-0703-11eb-9b20-419723c1a3b5.png)

**Elements that Violated it:**

`<div class="bx--search bx--search--xl table__toolbar__search" aria-labelledby="search__input__id_x85ie4ofski-search" role="search">`

**Rule Violated:** A <label> element must have non-empty descriptive text that identifies the purpose of the interactive component
![image](https://user-images.githubusercontent.com/17908476/95104177-8ace3c00-0703-11eb-990f-95fab35f0ce8.png)

**Elements that Violated it:**

`<label class="bx--label" for="search__input__id_x85ie4ofski" id="search__input__id_x85ie4ofski-search">`

**Comment by Pasoon** [comment]
**Sidenote:** When upgrading @carbon/ibm-security from 1.28.0 to ^1.28.0 latest, webpack-internal:///../node_modules/@carbon/ibm-security/es/components/Toolbar/Toolbar.js:246 Uncaught TypeError: Cannot read property 'ariaLabel' of undefined

**Comment by SimonFinney** [comment]
Thanks for raising @Pasoon ! Can you confirm that all of these violations are directly produced by component internals from `@carbon/ibm-security`?

On first glance, it appears that some of these violations may be application-specific - I'd love to reduce this list down so we have something targeted we can address as soon as possible.

**Comment by Pasoon** [comment]
Several ones under "Each form control must have an associated label" may be application specific but the accessibility checker mentions that "If the design cannot have a visible label, use the title attribute to provide a label". Even though we have done so the violation is still being shown. Do you have any other ones in mind?
